Index: pyasn1/codec/ber/decoder.py
===================================================================
RCS file: /cvsroot/pyasn1/pyasn1/pyasn1/codec/ber/decoder.py,v
retrieving revision 1.52
diff -u -r1.52 decoder.py
--- pyasn1/codec/ber/decoder.py	24 May 2012 16:25:22 -0000	1.52
+++ pyasn1/codec/ber/decoder.py	26 May 2012 13:24:48 -0000
@@ -200,7 +200,12 @@
         while index < substrateLen:
             subId = oct2int(substrate[index])
             index = index + 1
-            if subId > 127:
+            if subId == 128:
+                #ASN.1 spec forbids leading zeros (0x80) in sub-ID OID encoding,
+                #tolerating it opens a vulnerability.
+                #See http://www.cosic.esat.kuleuven.be/publications/article-1432.pdf page 7
+                raise error.PyAsn1Error('Invalid OID encoding - leading 0x80 for sub ID')
+            elif subId > 128:
                 # Construct subid from a number of octets
                 nextSubId = subId
                 subId = 0
